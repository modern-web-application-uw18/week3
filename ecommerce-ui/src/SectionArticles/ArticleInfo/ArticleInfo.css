.articleInfo .title {
  font-family: 'Play', sans-serif;
}
