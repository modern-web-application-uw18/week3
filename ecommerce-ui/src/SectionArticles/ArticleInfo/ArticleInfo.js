import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './ArticleInfo.css';

export default class ArticleInfo extends Component {

    render() {
      const { description, experiences } = this.props.article;
        return (
          <div className="articleInfo">
            <div className="title">
              {experiences.amount}+ {experiences.type}
            </div>
            <div className="text">
              {description}
            </div>
          </div>
        );
    }
}

ArticleInfo.propTypes = {
  article: PropTypes.shape({
    description: PropTypes.string.isRequired,
    experiences: PropTypes.shape({
      amount: PropTypes.number.isRequired,
      type: PropTypes.string.isRequired,
    }).isRequired
  }).isRequired,
}

ArticleInfo.defaultProps = {
  article: {
    description: '',
    experiences: {
      amount: 0,
      type: 'experiences'
    }
  },
}
