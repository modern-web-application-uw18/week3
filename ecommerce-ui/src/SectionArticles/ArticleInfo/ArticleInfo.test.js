import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import ArticleInfo from './ArticleInfo.js';

describe('ArticleInfo', () => {
  it('Smoke Test', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ArticleInfo />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Match Snapshot', () => {
    const component = TestRenderer.create(<ArticleInfo />);
    expect(component.toJSON()).toMatchSnapshot();
  });

})
