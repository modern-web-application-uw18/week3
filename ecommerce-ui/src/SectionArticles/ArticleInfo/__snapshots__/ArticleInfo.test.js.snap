// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleInfo Match Snapshot 1`] = `
<div
  className="articleInfo"
>
  <div
    className="title"
  >
    0
    + 
    experiences
  </div>
  <div
    className="text"
  >
    
  </div>
</div>
`;
